tsl/test/expected/bgw_job_stat_history.out

```diff
@@ -0,0 +1,168 @@
+$BODY$
+DECLARE
+    r RECORD;
+BEGIN
+    FOR i in 1..spins
+    LOOP
+        SELECT total_runs FROM _timescaledb_internal.bgw_job_stat WHERE job_id=job_param_id INTO r;
+        IF (r.total_runs = expected_runs) THEN
+            RETURN true;
+        ELSEIF (r.total_runs > expected_runs) THEN
+            RAISE 'num_runs > expected';
+        ELSE
+            PERFORM pg_sleep(0.1);
+        END IF;
+    END LOOP;
+    RAISE INFO 'wait_for_job_to_run: timeout after % tries', spins;
+    RETURN false;
+END
+$BODY$;
+CREATE PROCEDURE custom_job_ok(job_id int, config jsonb) LANGUAGE PLPGSQL AS
+$$
+BEGIN
+  RAISE INFO 'custom_job';
+END
+$$;
+CREATE PROCEDURE custom_job_error(job_id int, config jsonb) LANGUAGE PLPGSQL AS
+$$
+BEGIN
+  PERFORM 1/0;
+END
+$$;
+-- Do not log all jobs, only FAILED executions
+SHOW timescaledb.enable_job_execution_logging;
+ timescaledb.enable_job_execution_logging 
+------------------------------------------
+ off
+(1 row)
+
+-- Start Background Workers
+SELECT _timescaledb_functions.start_background_workers();
+ start_background_workers 
+--------------------------
+ t
+(1 row)
+
+SELECT pg_sleep(6);
+ pg_sleep 
+----------
+ 
+(1 row)
+
+SELECT add_job('custom_job_ok', '1h', initial_start := now()) AS job_id_1 \gset
+SELECT add_job('custom_job_error', '1h', initial_start := now()) AS job_id_2 \gset
+SELECT wait_for_job_to_run(:job_id_1, 1);
+ wait_for_job_to_run 
+---------------------
+ t
+(1 row)
+
+SELECT wait_for_job_to_run(:job_id_2, 1);
+ wait_for_job_to_run 
+---------------------
+ t
+(1 row)
+
+-- only 1 failure
+SELECT count(*), succeeded FROM timescaledb_information.job_history WHERE job_id >= 1000 GROUP BY 2 ORDER BY 2;
+ count | succeeded 
+-------+-----------
+     1 | f
+(1 row)
+
+SELECT proc_schema, proc_name, sqlerrcode, err_message FROM timescaledb_information.job_history WHERE job_id >= 1000 AND succeeded IS FALSE;
+ proc_schema |    proc_name     | sqlerrcode |   err_message    
+-------------+------------------+------------+------------------
+ public      | custom_job_error | 22012      | division by zero
+(1 row)
+
+SELECT _timescaledb_functions.stop_background_workers();
+ stop_background_workers 
+-------------------------
+ t
+(1 row)
+
+-- Log all executions
+ALTER SYSTEM SET timescaledb.enable_job_execution_logging TO ON;
+SELECT pg_reload_conf();
+ pg_reload_conf 
+----------------
+ t
+(1 row)
+
+\c :TEST_DBNAME :ROLE_SUPERUSER
+SELECT _timescaledb_functions.start_background_workers();
+ start_background_workers 
+--------------------------
+ t
+(1 row)
+
+SELECT pg_sleep(6);
+ pg_sleep 
+----------
+ 
+(1 row)
+
+SELECT scheduled FROM alter_job(:job_id_1, next_start => now());
+ scheduled 
+-----------
+ t
+(1 row)
+
+SELECT scheduled FROM alter_job(:job_id_2, next_start => now());
+ scheduled 
+-----------
+ t
+(1 row)
+
+SELECT wait_for_job_to_run(:job_id_1, 2);
```

Comment:
I checked the logfiles from the windows run, it seems the scheduler run into the problem at this point:

```
2024-03-20 19:32:22.654 GMT: super_user [5932] db_bgw_job_stat_history LOG:  00000: statement: ALTER SYSTEM SET timescaledb.enable_job_execution_logging TO ON;
[...]
2024-03-20 19:32:22.655 GMT: super_user [5932] db_bgw_job_stat_history LOG:  00000: statement: SELECT pg_reload_conf();
[...]
2024-03-20 19:32:22.659 GMT:  [2556]  LOG:  00000: background worker "User-Defined Action [1001]" (PID 2396) exited with exit code 1
[...]
2024-03-20 19:32:22.704 GMT:  [6188]  LOCATION:  _PG_init, D:\a\timescaledb\timescaledb\src\loader\loader.c:601
2024-03-20 19:32:22.722 GMT:  [6188]  ERROR:  XX000: unable to find job history 0
2024-03-20 19:32:22.722 GMT:  [6188]  LOCATION:  ts_bgw_job_stat_history_mark_end, D:\a\timescaledb\timescaledb\src\bgw\job_stat_history.c:171
[...]
2024-03-20 19:32:28.701 GMT: super_user [2884] db_bgw_job_stat_history LOG:  00000: statement: SELECT scheduled FROM alter_job(1000, next_start => now());
2024-03-20 19:32:28.703 GMT: super_user [2884] db_bgw_job_stat_history LOG:  00000: statement: SELECT scheduled FROM alter_job(1001, next_start => now());
[...]
2024-03-20 19:32:28.703 GMT: super_user [2884] db_bgw_job_stat_history LOG:  00000: statement: SELECT wait_for_job_to_run(1000, 2);
2024-03-20 19:32:28.703 GMT: super_user [2884] db_bgw_job_stat_history LOCATION:  exec_simple_query, d:\pginstaller_13.auto\postgres.windows-x64\src\backend\tcop\postgres.c:1043
2024-03-20 19:33:28.708 GMT: super_user [2884] db_bgw_job_stat_history ERROR:  57014: canceling statement due to statement timeout
```